validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonElementValidator.java

```diff
@@ -0,0 +1,135 @@
+        this.strict = strict;
+        this.context = context;
+        this.ignoreExpressions = ignoreExpressions;
+    }
+
+    public void validate(JsonElementValidatorItem<?> control) {
+        if (isIgnoredByPlaceholderOrExpressionList(ignoreExpressions, control))
+            return;
+        if (isValidationMatcherExpression(requireNonNullElse(control.expectedString(), ""))) {
+            resolveValidationMatcher(control.getJsonPath(), control.actualString(), control.expectedString(), context);
+        } else if (control.expected instanceof JSONObject) {
+            validateJSONObject(this, control);
+        } else if (control.expected instanceof JSONArray) {
+            validateJSONArray(this, control);
+        } else {
+            validateNativeType(control);
+        }
+    }
+
+    private void validateJSONObject(JsonElementValidator validator, JsonElementValidatorItem<?> control) {
+        var objectControl = control.ensureType(JSONObject.class);
+
+        if (strict) {
+            validateSameSize(objectControl.getJsonPath(), objectControl.expected.keySet(), objectControl.actual.keySet());
+        }
+
+        var controlEntries = objectControl.expected
+                .entrySet()
+                .stream()
+                .map(entry -> new JsonElementValidatorItem<>(
+                        entry.getKey(), objectControl.actual.get(entry.getKey()), entry.getValue()
+                ).parent(objectControl))
+                .toList();
+
+        for (var entryControl : controlEntries) {
+            if (!objectControl.actual.containsKey(entryControl.getName())) {
+                throw new ValidationException(buildValueToBeInCollectionErrorMessage(
+                        "Missing JSON entry", entryControl.getName(), objectControl.actual.keySet()
+                ));
+            }
+
+            validator.validate(entryControl);
+        }
+    }
+
+    /**
+     * Checks if given element node is either on ignore list or
+     * contains @ignore@ tag inside control message
+     */
+    public static boolean isIgnoredByPlaceholderOrExpressionList(Collection<String> ignoreExpressions, JsonElementValidatorItem<?> controlEntry) {
+        String trimmedControlValue = requireNonNullElse(controlEntry.expectedString(), "").trim();
+        if (trimmedControlValue.equals(IGNORE_PLACEHOLDER)) {
+            return true;
+        }
+
+        if (ignoreExpressions.stream().anyMatch(controlEntry::isPathIgnoredBy)) {
+            return true;
+        }
+
+        return false;
+    }
+
+
+    private void validateJSONArray(JsonElementValidator validator, JsonElementValidatorItem<?> control) {
+        var arrayControl = control.ensureType(JSONArray.class);
+        if (strict) {
+            validateSameSize(control.getJsonPath(), arrayControl.expected, arrayControl.actual);
+        }
+        for (int i = 0; i < arrayControl.expected.size(); i++) {
+            if (!isAnyValidItemInActualArray(validator, arrayControl, i)) {
```

Comment:
so just to be clear; strict mode does **not** strictly validate the order of elements - right? I think that is kinda okay, since JSON in general has no such restriction. on the other hand, someone might want to test it e.g. when the return something ordered (paginated) from their APIs. we would not allow them to do so with the current implementation. any thoughts?

Reply:
Yeah that's true, I also thought about that...

Options are:
1. Always ignore order
2. Ignore order if `strict = false` and enforce order if `strict = true`
3. Make `JsonElementValidator` more configurable with properties 
4. Make `JsonElementValidator` more configurable by easily allowing overriding of the element-validations (in the refactoring process, I thougt of a [Chain of Responsibility](https://refactoring.guru/design-patterns/chain-of-responsibility) where each link is one element validation... but found that too sophisticated for the moment)

(I fancy Option 1 or 2)

Reply:
so, option one is now implemented. I think that is acceptable. if someone really wants to check the order, he can easily implement a custom validator.

Reply:
seems to be a problem - addressed in https://github.com/citrusframework/citrus/pull/1155